### PR TITLE
add hotkeys for unslecting and deleting timespans

### PR DESCRIPTION
**Pull request by CalebBassham**
Added hotkeys for deleting timespans.

Selecting a timespan and pressing delete or backspace will delete it.
Pressing escape will unselect.

Pressing escape after creating a new timespan will delete the newly created timespan until it or another timespan has been selected. Side note: I think it would be preferred to set newly created timespans as selected and open the popper menu for tagging and whatnot and this side case would no longer be necessary as just pressing delete and backspace / the trash icon would feel more natural. But... I can't figure out how to get that to work and without it, this feels pretty good.

### CODE REVIEW

File: ui/src/timespan/TimeSpan.tsx [hunk 1]
```diff
@@ -16,14 +16,13 @@ import Menu from '@material-ui/core/Menu';
 import MenuItem from '@material-ui/core/MenuItem';
 import {RemoveTimeSpan, RemoveTimeSpanVariables} from '../gql/__generated__/RemoveTimeSpan';
 import {useStateAndDelegateWithDelayOnChange} from '../utils/hooks';
-import {TimeSpans} from '../gql/__generated__/TimeSpans';
 import {isSameDate} from '../utils/time';
```
Comment (jmattheis) [inline ui/src/timespan/TimeSpan.tsx:19]: > Side note: I think it would be preferred to set newly created timespans as selected and open the popper menu for tagging and whatnot and this side case would no longer be necessary as just pressing delete and backspace / the trash icon would feel more natural. But... I can't figure out how to get that to work and without it, this feels pretty good.

I'm okay with that, it could probably work with the following steps. But I think I rather not do this because it's hacky:
- save the timespan id of the newly created timespan
- wait for full calender rerender
- search the dom for the newly created element for the timespan
- find the timespan object by id
- call setSelected() with the found data

Basically the main problem is that the onSelect event from the calender doesn't provide the newly created element and only has the full calender as event target.

File: ui/src/timespan/TimeSpan.tsx [hunk 2]
```diff
@@ -16,14 +16,13 @@ import Menu from '@material-ui/core/Menu';
 import MenuItem from '@material-ui/core/MenuItem';
```
Comment (jmattheis) [inline ui/src/timespan/TimeSpan.tsx:16]: > Selecting a timespan and pressing delete or backspace will delete it.
> Pressing escape will unselect.

> Pressing escape after creating a new timespan will delete the newly created timespan until it or another timespan has been selected. 

In my testing backspace/delete works too for newly created timespans (by dragging). Why does we also need to support escape here? I rather not have escape mean sometimes delete and sometimes deselect.

File: ui/src/timespan/TimeSpan.tsx [hunk 3]
```diff
@@ -16,14 +16,13 @@ import Menu from '@material-ui/core/Menu';
...
 import {RemoveTimeSpan, RemoveTimeSpanVariables} from '../gql/__generated__/RemoveTimeSpan';
 import {useStateAndDelegateWithDelayOnChange} from '../utils/hooks';
```
Comment (jmattheis) [inline ui/src/timespan/TimeSpan.tsx:18]: Mhmm, I think travis ci finally killed the open source builds or this project doesn't have credits left.

**Review by jmattheis — CHANGES_REQUESTED** [review]
Thanks for the contributes, I've added remarks in sub comments.

File: ui/src/timespan/TimeSpan.tsx (hunk 1 above)
Comment (CalebBassham) [reply ui/src/timespan/TimeSpan.tsx:19]: Yea, that is the issue I was running into. That is a bit of a pain. It can just stay how it is.

File: ui/src/timespan/TimeSpan.tsx (hunk 3 above)
Comment (CalebBassham) [reply ui/src/timespan/TimeSpan.tsx:18]: Tragic. Probably time to move to GitHub Actions though anyway.

File: ui/src/timespan/calendar/CalendarPage.tsx [hunk 4]
```diff
@@ -93,18 +100,34 @@ export const CalendarPage: React.FC = () => {
         window.__TRAGGO_CALENDAR = {};
         return () => (window.__TRAGGO_CALENDAR = undefined);
     });
+
+    const fullCalendarRef = React.useRef<FullCalendar | null>(null);
+    const unselect = () => {
```
[the remark this answers is not in the document]
Comment (CalebBassham) [reply ui/src/timespan/calendar/CalendarPage.tsx:105]: For sure. 

I think this is out of the scope of this pr, but as an outside coming in, it is quite confusing and a general pass over this entire section of code and explicitly changing variables and functions to reference if it is the selected event (what this project calls TimeSpan) or the selection (what fullcalendar calls it, but maybe call it calendar selection or selected time range or something to differentiate.)

File: ui/src/timespan/TimeSpan.tsx (hunk 2 above)
Comment (CalebBassham) [reply ui/src/timespan/TimeSpan.tsx:16]: It just feels really intuitive in my opinion and a more convenient key press for quick mistakes.

**Review by jmattheis — APPROVED** [review]
I fixed some linting errors, otherwise :+1:
